test/interactions/interactionTests.ts

```diff
@@ -0,0 +1,94 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactions", () => {
+  describe("Interactions", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    it("can attach/detach interaction to/from component", () => {
+      var svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component = new Plottable.Component();
+      component.renderTo(svg);
+
+      var clickInteraction = new Plottable.Interactions.Click();
+
+      var callbackCalled = false;
+      var callback = () => callbackCalled = true;
+      clickInteraction.onClick(callback);
+
+      clickInteraction.attachTo(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "callback called on clicking Component (mouse)");
+
+      callbackCalled = false;
+      clickInteraction.detachFrom(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "callback was removed from component and should not be called");
+
+      svg.remove();
+    });
+
+    it("interactions are reusable", () => {
+      var svg1 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var svg2 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component1 = new Plottable.Component();
+      var component2 = new Plottable.Component();
+
+      component1.renderTo(svg1);
+      component2.renderTo(svg2);
+
+      var clickInteraction = new Plottable.Interactions.Click();
+
+      var callbackCalled: Boolean;
```

Comment:
We should probably tslint-ban the non-primitive `Boolean`, `Number`, and `String` types until we can think of a good use for them.